src/main/java/entity/CommonTaskFactoryInterface.java

```diff
@@ -0,0 +1,6 @@
+package entity;
+
+public interface CommonTaskFactoryInterface {
```

Comment:
I don't think this file is needed

Reply:
You are correct, will remove it when I push my next changes

Reply:
Should use the general taskfactory as interface, and the commontaskfactory as the concrete object in the constructor when you are using the interactor